frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/print/SimpleGraphCellPainter.java

```diff
@@ -0,0 +1,189 @@
+package com.virtuslab.gitmachete.frontend.graph.print;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.geom.Ellipse2D;
+import java.util.Collection;
+
+import com.intellij.vcs.log.graph.EdgePrintElement;
+import com.intellij.vcs.log.graph.NodePrintElement;
+import com.intellij.vcs.log.graph.PrintElement;
+import com.intellij.vcs.log.paint.ColorGenerator;
+import com.intellij.vcs.log.paint.GraphCellPainter;
+import com.intellij.vcs.log.paint.PaintParameters;
+import com.intellij.vcs.log.paint.PositionUtil;
+import lombok.RequiredArgsConstructor;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+@RequiredArgsConstructor
+public class SimpleGraphCellPainter implements GraphCellPainter {
+
+  private final ColorGenerator myColorGenerator;
+
+  @UIEffect
+  protected int getRowHeight() {
+    return PaintParameters.ROW_HEIGHT;
+  }
+
+  @UIEffect
+  private BasicStroke getOrdinaryStroke() {
+    return new BasicStroke(PaintParameters.getLineThickness(getRowHeight()), BasicStroke.CAP_ROUND,
+        BasicStroke.JOIN_BEVEL);
+  }
+
+  @UIEffect
+  private void paintUpLine(Graphics2D g2, Color color, int from, int to) {
```

Comment:
Hmm it's unclear what `from` and `to` refer to... graph node indexes? Same for other methods

Reply:
Change, rereview please :pray: